### PR TITLE
SHOP-1747  new approach for serverConfig.apollo.serverOptions

### PR DESCRIPTION
**Pull request by vikas-mindpath**
[SHOP-1747](https://withjoy.atlassian.net/browse/SHOP-1747)

B: server-core: new approach for 'serverConfig.apollo.serverOptions'

[SHOP-1747]: https://withjoy.atlassian.net/browse/SHOP-1747?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/middleware/error.logging.ts [hunk 1]
```diff
@@ -41,8 +41,7 @@ export const errorLoggingExpress: ErrorRequestHandler = function errorLoggingExp
     query: _loggedNonEmpty(req.query),
     body: _loggedNonEmpty(req.body),
   }, identity); // eg. `compact`
-
-  const { statusCode } = (err as any); // "Property 'FOO' does not exist on type 'Error'."
+  const { statusCode } = ((err as any).extensions?.props)? (err as any).extensions.props : (err as any); // "Property 'FOO' does not exist on type 'Error'."
```
Comment (vikas-mindpath) [inline src/middleware/error.logging.ts:44]: New way to handle error

Comment (cantremember) [inline src/middleware/error.logging.ts:44]: the line has gotten too long

```suggestion
  // "Property 'FOO' does not exist on type 'Error'."
  const { statusCode } = ((err as any).extensions?.props)? (err as any).extensions.props : (err as any); 
```

File: src/server/apollo.config.spec.ts [hunk 2]
```diff
@@ -369,15 +366,13 @@ describe('server/apollo.config', () => {
...
-            mocks: false,
-            playground: false,
+            cacheControl: false,
             reporting: false,
-            subscriptions: false,
             tracing: true,
+            introspection: true,
+            allowBatchedHttpRequests: true,
+            stopOnTerminationSignals: true,
+            includeStacktraceInErrorResponses: false,
```
Comment (cantremember) [inline src/server/apollo.config.spec.ts:375]: 👌

File: src/server/apollo.server.ts [hunk 3]
```diff
@@ -3,15 +3,27 @@ import { ErrorRequestHandler } from 'express';
...
-
+  if (args.cacheControl === false) {
+    plugins.push(ApolloServerPluginCacheControlDisabled());
+  }
+  if (args.reporting === false) {
+    plugins.push(ApolloServerPluginUsageReportingDisabled())
+  }
+  if (args.tracing === false) {
+    plugins.push(ApolloServerPluginInlineTraceDisabled())
+  }
```
Comment (cantremember) [inline src/server/apollo.server.ts:26]: 💯

**Review by cantremember — APPROVED** [review]
## one superficial suggestion

then ✅ and merge it

**Pull request merged**